trace/trace.go

```diff
@@ -571,6 +575,8 @@ type defaultIDGenerator struct {
 
 	traceIDAdd  [2]uint64
 	traceIDRand *rand.Rand
+
+	randomIDsGenerated bool
```

Comment:
this field is redundant if you introduce `initOnce sync.Once` instead like I suggested above.